flash/core/integrations/icevision/transforms.py

```diff
@@ -239,9 +240,12 @@ def __init__(self, transform: List[Callable]):
         self.transform = A.Adapter(transform)
 
     def forward(self, x):
+
         record = to_icevision_record(x)
         record = self.transform(record)
-        return from_icevision_record(record)
+        record = from_icevision_record(record)
+        record[DataKeys.METADATA]["original_size"] = x[DataKeys.INPUT].shape[:-1]
```

Comment:
This should probably live within the input object rather than in the transform adapter as the adapter may be used multiple times